### PR TITLE
manage: add apikey management

### PR DESCRIPTION
**Pull request by liewegas**
Not sure if the last commit (use_count and last_use) is a good idea or not. It makes the key management easier!

### CODE REVIEW

File: app/apikey/manage_views.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            Client, default_slug__slug=kwargs["subscriber"]
+        )
+        if request.user.is_authenticated and (
+            self.subscriber != request.user.active_client
+            or self.subscriber not in request.user.allowed_clients
+        ):
+            raise Http404
+        if not self.subscriber.has_api_access:
+            raise Http404
+        return super().dispatch(request, *args, **kwargs)
```
Comment (liewegas) [inline app/apikey/manage_views.py:38]: Is there a better way to do this?  I want SubscriberManageViewMixin (which loads at sets ``self.subscriber``) + a check that ``self.subscriber.has_api_access``, but if I overload then subscriber isn't set until after I call super().dispatch(...)...

File: app/multi_tenant/migrations/0018_auto_20200803_2031.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        ),
+        migrations.AlterField(
+            model_name='client',
+            name='email',
+            field=models.EmailField(default='info@voteamerica.com', max_length=254, null=True, verbose_name="Default 'From' Email Address"),
+        ),
+        migrations.AlterField(
+            model_name='client',
+            name='privacy_policy',
+            field=models.URLField(blank=True, null=True, verbose_name='Privacy Policy URL'),
```
Comment (benweissmann) [inline app/multi_tenant/migrations/0018_auto_20200803_2031.py:26]: Any idea what's up with all these `AlterField`s?

File: app/multi_tenant/models.py [hunk 3]
```diff
@@ -59,6 +59,8 @@ class Client(UUIDModel, TimestampModel):
     # this client.
     is_first_party = models.BooleanField(default=False)
 
+    has_api_access = models.BooleanField(default=False)
```
Comment (benweissmann) [inline app/multi_tenant/models.py:62]: I messed this up with `is_first_party`, but Nick generally wants everything to be `null=True` to make red/back deployments easier / make running the migrations a bit more independent of updating the code

File: app/apikey/manage_views.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    UpdateView,
+)
+
+from manage.mixins import ManageViewMixin
+from multi_tenant.models import Client
+
+from .forms import ApiKeyCreateForm, ApiKeyDeactivateForm
+from .models import ApiKey
+
+logger = logging.getLogger("api_crypto")
```
[the remark this answers is not in the document]
Comment (liewegas) [reply app/apikey/manage_views.py:22]: Or ``apikey`` for both this and in crypto.py?  I never seem to see any log output until I add the logger to the ``LOGGING`` defining in settings... but maybe that is only true for local development?

[the remark this answers is not in the document]
Comment (benweissmann) [reply app/apikey/manage_views.py:22]: Yeah, apikey for both seems good. I'm not... totally sure how the logging configuration works; I definitely forgot to add it to the `LOGGING` configuration when I added this logger. Maybe there's some way to set a default?

File: app/apikey/models.py [hunk 5]
```diff
@@ -30,11 +32,20 @@ class ApiKey(SubscriberModel, UUIDModel, TimestampModel):
...
 
     def __str__(self):
-        return f"API Key for {self.subscriber} - {self.pk}"
+        return f"API Key for {self.subscriber} - {self.description} - {self.pk}"
 
     def hashed_secret(self) -> str:
         return hash_key_secret(self.secret)
+
+    def note_use(self) -> None:
+        self.use_count = F("use_count") + 1
```
[the remark this answers is not in the document]
Comment (liewegas) [reply app/apikey/models.py]: I'll leave this off for now

**Pull request merged**